pkg/provider/loadbalancer/accesscontrol.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loadbalancer
+
+import (
+	"errors"
+	"fmt"
+	"net/netip"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/cloud-provider-azure/pkg/consts"
+)
+
+var (
+	ErrNotLoadBalancer = errors.New("expect service to be LoadBalancer")
+)
+
+// IsInternal returns true if the given service is internal load balancer.
+func IsInternal(svc *v1.Service) bool {
```

Comment:
It would be simpler if we put it in provider package so that we can use the existing functions such as requireInternalLoadBalancer.

Reply:
I suggest that we split the abstraction or implementation into smaller, neat packages. Grouping everything into a single scope/package, or attaching it to a massive structure such as provider.Cloud, would only increase the complexity. It seems equivalent to the use of global variable.